### DISCUSSION TITLE
alda-player process initialize slowly

### DISCUSSION THREAD

**Discussion by yueyoum** [alda-lang/alda#498] [Q&A]
```
OS:  Win10
Terminal:  CygWin
JAVA: 
$ java --version
java 21.0.2 2024-01-16 LTS
Java(TM) SE Runtime Environment (build 21.0.2+13-LTS-58)
Java HotSpot(TM) 64-Bit Server VM (build 21.0.2+13-LTS-58, mixed mode, sharing)

SoundFound:   207MB MuseScore_General.sf2
```

When I run `alda repl` for the first time,  or at the momet no player process existing. 
It will consume 100% CPU for almost 30s.  
At this time,  alda repl shell has no response .

![alda_20240325152946](https://github.com/alda-lang/alda/assets/2081686/8c541b8c-e0fa-4687-960d-15879120abb2)

Even it report a WARN msg:

![alda_warn](https://github.com/alda-lang/alda/assets/2081686/33162529-181d-4510-b835-e594865b270b)

```
Starting player processes...
Mar 25 15:29:36 WRN repl/player_management.go:162 > No player processes availabl
e. error="It looks like Alda is having trouble starting player processes in the\
nbackground. This could happen for a number of reasons.\n\nTo troubleshoot:\n\n
 • Run \x1b[93malda doctor\x1b[0m and see if any of the health checks fail.\n\n
 • Run \x1b[93malda ps\x1b[0m to see the state of any currently running player\n
    processes.\n\n  • Look for error messages in:\n      \x1b[93mC:\\Users\\wang
\\AppData\\Local\\alda\\cache\\logs\\alda-player.log\x1b[0m\n\n  • Try running a
 player process in the foreground:\n      \x1b[93malda-player -v run -p 27278\x1
b[0m\n\n  • Try to make it play something:\n      \x1b[93malda -v2 play -p 27278
 -c \"piano: c12 e g > c4\"\x1b[0m"

```

Sometimes when first start the alda repl shell,
I quickly type `piano c d e f` and press Enter

No sound happens, but wait for a while.

This is a bad experience. So that I wanna a long running player process before. 

Is this due to wait the player process  ready/active ?
and Why initialize the player process so slowly ?  (big sf2 file ?)

**Comment by daveyarwood** [comment]
I'm not sure, off hand, what the issue could be, but it might be related to this: https://github.com/alda-lang/alda/issues/410

I think if we adjusted our JVM options, we could make the player processes more lightweight. But I'm not very knowledgeable in that area.
